cola/widgets/diff.py

```diff
@@ -839,6 +855,10 @@ def process_diff_selection(self, reverse=False, apply_to_worktree=False):
                 first_line_idx, last_line_idx,
                 self.has_selection(), reverse, apply_to_worktree)
 
+    def _cursor_changed(self):
+        super(DiffEditor, self)._cursor_changed()
+        self.selection_model.line_number = self.numbers.newfile_line()
+
```

Comment:
Last tweak -- if we're going to override `_cursor_changed` from the base class then we should rename it to `cursor_changed` (without the leading underscore) to signal that it's a a little more "public".  I try to avoid touching underscore things outside of the same class.